lib/PermissionManager.php

```diff
@@ -64,7 +64,7 @@ private function userMatchesGroupList(?string $userId = null, ?array $groupList
 		}
 
 		if ($userId === null) {
-			return true;
+			return false;
```

Comment:
Maybe add an explicit comment there to explain how unauthenticated users are handled.

Reply:
Definitely, added :)